models.py

```diff
@@ -88,9 +98,35 @@ def zulip_link(self):
         url = zulip_util.stream_topic_to_narrow_url(self.stream, self.subject)
         return "**[#{} > {}]({})**".format(self.stream, self.subject, url)
 
+@sqlalchemy.event.listens_for(Event, 'before_insert')
+def ensure_one_event_per_thread(mapper, conn, event):
+    if event.already_initialized():
+        thread_already_taken = Session.query(Event).filter(Event.stream == event.stream).filter(Event.subject == event.subject).count() > 0
```

Comment:
I think this can use `event_exists`. Maybe put that in here and then import it in rsvp_commands.py?